src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,597 @@
+    uint maxTotalCostTolerance;
+    /// @dev used as tolerance for how much TVL we could possibly lose from one RFQ
+    uint maxLossOrGainPercentOfTVL;
+    /// @dev the max tolerance we allow when calculating cost of a trade
+    uint negMaxCashTolerance;
+    /// @dev Minimum time before an action is expired
+    uint minSignatureExpiry;
+    /// @dev Maximum time before an action is expired
+    uint maxSignatureExpiry;
+    /// @dev Lower bound for option expiry
+    uint optionMinTimeToExpiry;
+    /// @dev Upper bound for option expiry
+    uint optionMaxTimeToExpiry;
+    /// @dev Maximum amount of negative cash allowed to be held to open any more option positions. (e.g. -100e18)
+    int maxNegCash;
+    /// @dev Percentage of an rfq price that can be taken as a fee.
+    uint rfqFeeFactor;
+  }
+
+  /// @custom:storage-location erc7201:lyra.storage.PrincipalProtectedTSA
+  struct PPTSAStorage {
+    IDepositModule depositModule;
+    IWithdrawalModule withdrawalModule;
+    ITradeModule tradeModule;
+    IRfqModule rfqModule;
+    IOptionAsset optionAsset;
+    ISpotFeed baseFeed;
+    PPTSAParams ppParams;
+    CollateralManagementParams collateralManagementParams;
+    bool isCallSpread;
+    bool isLongSpread;
+    /// @dev Only one hash is considered valid at a time, and it is revoked when a new one comes in.
+    bytes32 lastSeenHash;
+  }
+
+  struct StrikeData {
+    uint strike;
+    uint expiry;
+    int markPrice;
+    uint tradePrice;
+    int tradeAmount;
+  }
+
+  // keccak256(abi.encode(uint256(keccak256("lyra.storage.PrincipalProtectedTSA")) - 1)) & ~bytes32(uint256(0xff))
+  bytes32 private constant PPTSAStorageLocation = 0x273fadd8b9727c49bd8eabddca41871e1e7fc468f40ece57b00a1bd25002e500;
+
+  function _getPPTSAStorage() private pure returns (PPTSAStorage storage $) {
+    assembly {
+      $.slot := PPTSAStorageLocation
+    }
+  }
+
+  constructor() {
+    _disableInitializers();
+  }
+
+  function initialize(
+    address initialOwner,
+    BaseTSA.BaseTSAInitParams memory initParams,
+    PPTSAInitParams memory ppInitParams
+  ) external reinitializer(2) {
+    __BaseTSA_init(initialOwner, initParams);
+
+    PPTSAStorage storage $ = _getPPTSAStorage();
+
+    $.depositModule = ppInitParams.depositModule;
+    $.withdrawalModule = ppInitParams.withdrawalModule;
+    $.rfqModule = ppInitParams.rfqModule;
+    $.tradeModule = ppInitParams.tradeModule;
+    $.optionAsset = ppInitParams.optionAsset;
+    $.baseFeed = ppInitParams.baseFeed;
+    $.isCallSpread = ppInitParams.isCallSpread;
+    $.isLongSpread = ppInitParams.isLongSpread;
+    BaseTSAAddresses memory tsaAddresses = getBaseTSAAddresses();
+
+    tsaAddresses.depositAsset.approve(address($.depositModule), type(uint).max);
+  }
+
+  ///////////
+  // Admin //
+  ///////////
+  function setPPTSAParams(CollateralManagementParams memory newCollateralMgmtParams, PPTSAParams memory pptsaParams)
```

Comment:
would use a seperate setter for collateralManagementParams, and keep it in CollateralManagementTSA